packages/application/src/requests.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+
+import * as React from 'react';
+
+import { ErrorDialogContent } from './ErrorDialogContent';
+
+const HTML_STATUS_SUCCESS = [200, 201];
```

Comment:
HTTP response status codes indicate whether a specific HTTP request has been successfully completed. Responses are grouped in five classes:

- Informational responses (100–199),
- Successful responses (200–299),
- Redirects (300–399),
- Client errors (400–499),
- Server errors (500–599).


Within successful responses these are defined on the RFC 

- 200 ok
- 201 created
- 202 accepted
- 203 Non-Authoritative information
- 204 No Content
- 205 Reset Content
- 206 Partial Content
and a few other WebDAV related

Reply:
I figured we could add more valid cases here as we have them. Currently these are the only two valid success statuses that the elyra api returns. I can make this check for any value in the 200–299 range instead, but I though it'd be better to be specific because if we get a "success" code that shouldn't be possible we should treat it as an error.

Reply:
I would prefer that the accepted range be (200-299) but would be ok with (200-2010) for now. @kevin-bates thoughts here? what are the success status code range you have been using?

Reply:
We're only using 200 and 201.  204 (No Content) could be used on delete if we were to choose not to return the deleted resource, but since we return content we use 200 on deletes.

Reply:
though I would normally want to only handle expected cases as success, since this is a util I will update this to handle all 200-299 as success to follow html standard